src/components/basic_table/basic_table.tsx

```diff
@@ -690,14 +690,29 @@ export class EuiBasicTable<T = any> extends Component<
 
       // field data column
       const sorting: SortOptions = {};
-      if (this.props.sorting && sortable) {
-        const sortDirection = this.resolveColumnSortDirection(column);
-        sorting.isSorted = !!sortDirection;
-        sorting.isSortAscending = sortDirection
-          ? SortDirection.isAsc(sortDirection)
-          : undefined;
-        sorting.onSort = this.resolveColumnOnSort(column);
-        sorting.allowNeutralSort = this.props.sorting.allowNeutralSort;
+      if (this.props.sorting) {
+        if (
+          this.props.sorting.sort &&
+          !!this.props.sorting.enableAllColumns &&
+          !(column as EuiTableFieldDataColumnType<T>).sortable
```

Comment:
Need to check explicitly for non-existance of the field, instead of falsey, so that a value of `false` can be used to disable sorting on a column.

```suggestion
          (column as EuiTableFieldDataColumnType<T>).sortable == null
```